src/ParameterLists.f90

```diff
@@ -2054,13 +2200,14 @@ SUBROUTINE getString_ParamType_scalar(thisParam,name,string,sskfmt,sdkfmt)
 !> @param name
 !> @param hasname
 !>
-    SUBROUTINE getString_ParamType_a1(thisParam,name,string,sskfmt,sdkfmt)
+    SUBROUTINE getString_ParamType_a1(thisParam,name,string,sskfmt,sdkfmt,delim)
```

Comment:
Why were the converters of arrays implemented in the routine for scalar? Are these supposed to go to the following routines?

Reply:
These routines are the original set of procedures for the generic `%getString` interface.  Originally, I envisioned doing a consistent dimensional conversion from an arbitrary intrinsic type to the corresponding string type (e.g. scalar int to scalar string, 1-D float array to 1-D string array, and so on...)  For the conversion of parameter list to output table, I wanted to be able to put a 1-D array into a single table entry, so I needed to modify `%getString_ParamType` to handle the conversion of multidimensional types to a single scalar string (e.g. scalar int to scalar string, 1-D float array to scalar string, and so on...).

The only change here is adding an optional delimiter, just so the procedures for the interfaces were consistent and simpler to understand.  The delimiter isn't used in this or other higher dimensional routines.  So, keep for consistency, or remove because not used?

Reply:
Oh..that makes sense. Maybe remove 'delim' for non-scalar routines? That way compiler will complain if you call it with 'delim' but accidentally pass a multi-dimentional string, instead of silently converting it using the non-scalar routines.

Reply:
Sounds good to me.  I'll make the change.